receiver/pluginreceiver/plugin.go

```diff
@@ -91,6 +100,32 @@ func (p *Plugin) ApplyDefaults(values map[string]any) map[string]any {
 	return result
 }
 
+// CheckExtensions checks to see if there exists any filestorage extensions
+// within the plugin config. If one exists, make a unique storage extension
+// directory based on the receiver component ID.
+func CheckExtensions(values map[string]any, id string) error {
+	for i, ext := range values {
+		if i == "file_storage" {
```

Comment:
This should be a [strings.HasPrefix](https://pkg.go.dev/strings#HasPrefix) as it's valid to have something like `file_storage/one`.

Reply:
I think the *IDEAL* thing to do here is to use https://github.com/open-telemetry/opentelemetry-collector/blob/b1e161415ab7838b82fb5336a6f52b61d914bd4f/config/identifiable.go#L57 and use the `Type` method to get the type; This way we know that it matches OTel parsing exactly.